frontend/packages/pipelines-plugin/src/components/pipelines/utils/triggers.ts

```diff
@@ -28,7 +28,7 @@ type TriggerTemplateMapping = { [key: string]: TriggerTemplateKind };
 
 const getResourceName = (resource: K8sResourceCommon): string => resource.metadata.name;
 const getEventListenerTemplateNames = (el: EventListenerKind): string[] =>
-  el.spec.triggers.map((elTrigger: EventListenerKindTrigger) => elTrigger.template?.name);
+  el.spec?.triggers?.map((elTrigger: EventListenerKindTrigger) => elTrigger.template?.name) || [];
```

Comment:
Nit: `spec` is not optional.
```suggestion
  el.spec.triggers?.map((elTrigger: EventListenerKindTrigger) => elTrigger.template?.name) || [];
```